src/System.Net.Primitives/src/System/Net/IPAddressParser.cs

```diff
@@ -213,19 +213,20 @@ public static unsafe bool Ipv6StringToAddress(ReadOnlySpan<char> ipSpan, ushort*
                 long result = 0;
                 if (!string.IsNullOrEmpty(scopeId))
                 {
-                    if (scopeId.Length < 2)
+                    uint interfaceIndex = Interop.Sys.GetNativeIPInterfaceIndex(scopeId);
+                    if (interfaceIndex >= 0)
                     {
-                        scope = 0;
-                        return false;
+                        scope = interfaceIndex;
+                        return true;
                     }
 
-                    for (int i = 1; i < scopeId.Length; i++)
+                    for (int i = 0; i < scopeId.Length; i++)
```

Comment:
Should this loop be replaced by uint.TryParse instead?

Reply:
Done